### DISCUSSION TITLE
How many groups (2349*?) does windowpost support within one deadline? Can more than 2 be submitted in time?

### DISCUSSION THREAD

**Discussion by lbj2004032** [filecoin-project/lotus#9100] [General Q&A] [answered]
I know that lotus has 2349 blocks in one group. If it exceeds, it will be divided into multiple groups for windowpost. My question is if there are 2349*3 blocks in one of my deadlines, can lotus still submit windowpost in time? There is no problem with grouping, but is there a problem if there are more than 2?

**Comment by rjan90** [comment, marked as answer]
My answer from the issue you also created:

> When you get to multiple partitions, especially when there is more then 2 partitions, it is time to think about setting up windowPoSt workers: https://lotus.filecoin.io/storage-providers/seal-workers/post-workers/#multiple-partitions. If setting up multiple windowPoSt-workers, each worker will compute one or more partitions (depending on how many partitions you have) 😄

**Reply by rjan90** [reply]
Just adding also that when you get above 2-3 partitions in a deadline, you might consider splitting the max partitions per windowPoSt and DeclareFaultsRecovered messages:

```
  # The maximum number of sectors which can be proven in a single PoSt message is 25000 in network version 16, which
  # means that a single message can prove at most 10 partinions
  # 
  # In some cases when submitting PoSt messages which are recovering sectors, the default network limit may still be
  # too high to fit in the block gas limit; In those cases it may be necesary to set this value to something lower
  # than 10; Note that setting this value lower may result in less efficient gas use - more messages will be sent,
  # to prove each deadline, resulting in more total gas use (but each message will have lower gas limit)
  # 
  # Setting this value above the network limit has no effect
  #
  # type: int
  # env var: LOTUS_PROVING_MAXPARTITIONSPERMESSAGE
  #MaxPartitionsPerMessage = 0
  
  # In some cases when submitting DeclareFaultsRecovered messages,
  # there may be too many recoveries to fit in a BlockGasLimit.
  # In those cases it may be necessary to set this value to something low (eg 1);
  # Note that setting this value lower may result in less efficient gas use - more messages will be sent than needed,
  # resulting in more total gas use (but each message will have lower gas limit)
  #
  # type: int
  # env var: LOTUS_PROVING_MAXPARTITIONSPERRECOVERYMESSAGE
  #MaxPartitionsPerRecoveryMessage = 0
```
